cpu/atmega_common/periph/gpio_ll_irq.c

```diff
@@ -49,32 +51,52 @@ static struct isr_ctx isr_ctx[GPIO_EXT_INT_NUMOF];
 
 static void clear_pending_irqs(uint8_t exti)
 {
+#if defined(EIFR)
     EIFR |= 1 << exti;
+#elif defined(GIFR)
+    GIFR |= 1 << (INTF0 + exti);
+#endif
```

Comment:
```suggestion
#if defined(EIFR)
    EIFR |= 1 << exti;
#else
    GIFR |= 1 << (INTF0 + exti);
#endif
```

Why not like this? Any MCU variant that has neither `EIFR` nor `GIRF` would need another extension of the function, and IMO it is better if this fails to compile than causing issues at runtime.

Alternatively, if you want the error message to be more helpful:

```suggestion
#if defined(EIFR)
    EIFR |= 1 << exti;
#elif defined(GIFR)
    GIFR |= 1 << (INTF0 + exti);
#else
#  error "No support for AVR with neither EIFR nor GIFR"
#endif
```